mirascope/base/types.py

```diff
@@ -186,3 +186,79 @@ def content(self) -> str:
         the empty string.
         """
         ...  # pragma: no cover
+
+
+class BaseChunkerParams(BaseModel):
+    """The parameters with which to make a chunker."""
+
+    chunk_size: Optional[int] = None
+    chunk_overlap: Optional[int] = None
+    length_function: Optional[Callable[[str], int]] = None
+    keep_separator: bool = False
+    add_start_index: bool = False
+    strip_whitespace: bool = True
+
+    model_config = ConfigDict(extra="allow", arbitrary_types_allowed=True)
+
+    def kwargs(
+        self,
+        exclude: Optional[set[str]] = None,
+    ) -> dict[str, Any]:
+        """Returns all parameters for the embedder as a keyword arguments dictionary."""
+        extra_exclude = {"tools", "weave"}
+        exclude = extra_exclude if exclude is None else exclude.union(extra_exclude)
+        kwargs = {
+            key: value
+            for key, value in self.model_dump(exclude=exclude).items()
+            if value is not None
+        }
+        return kwargs
+
+
+class BaseEmbeddingParams(BaseModel):
+    """The parameters with which to make an embedding."""
+
+    model: str
+    model_config = ConfigDict(extra="allow", arbitrary_types_allowed=True)
+
+    def kwargs(
+        self,
+        exclude: Optional[set[str]] = None,
+    ) -> dict[str, Any]:
+        """Returns all parameters for the embedder as a keyword arguments dictionary."""
+        extra_exclude = {"tools", "weave"}
```

Comment:
unnecessary for these call params, right?

Reply:
if we use openai embedder, we should have weave?

Reply:
These are fields of `BaseCallParams` that don't exist on this object.

The `weave` field gets updated if you use the `with_weave` decorator, which currently supports `BaseCall` and `BaseExtractor` classes.

We should likely add support for `with_weave` with all of our rag classes. I'm happy to take care of that once we get the initial feature for RAG merged in.

Reply:
okay i will remove for now